src/algebra/lie/basic.lean

```diff
@@ -986,6 +1104,32 @@ end quotient
 
 end lie_submodule
 
+section lie_module
+
+variables (R : Type u) (L : Type v) (M : Type w)
+variables [comm_ring R] [lie_ring L] [lie_algebra R L] [add_comm_group M] [module R M]
+variables [lie_ring_module L M] [lie_module R L M]
+
+namespace lie_module
+
+/-- The derived Lie submodule of a Lie module. -/
+def derived_lie_submodule : lie_submodule R L M := ⁅(⊤ : lie_ideal R L), ⊤⁆
```

Comment:
The full name looks like `lie_module.derived_lie_submodule`. Maybe this should just be `lie_submodule.derived`?

Reply:
I'm happy to make a change but I'm a bit unsure about `lie_submodule.derived`.

This is the derived Lie submodule of a Lie module. Am I silly to think that the bit to the left of the `.` is the thing from which we derive and the bit to the right is the thing we get when we derive?

How would you feel about something in the room namespace called `derived_lie_submodule`?

Reply:
My motivation was that you could read `lie_submodule.derived` as "the `lie_submodule` which is `derived`". But `derived_lie_submodule` in the root namespace sounds fine too :+1:

Reply:
I've gone for `derived_lie_submodule` in the root.